src/WingetCreateCLI/Commands/ShowCommand.cs

```diff
@@ -0,0 +1,212 @@
+            }
+
+            List<string> latestManifestContent;
+
+            try
+            {
+                latestManifestContent = await this.GitHubClient.GetManifestContentAsync(this.Id, this.Version);
+                Manifests originalManifests = Serialization.DeserializeManifestContents(latestManifestContent);
+                this.ParseArgumentsAndShowManifest(originalManifests);
+                return await Task.FromResult(commandEvent.IsSuccessful = true);
+            }
+            catch (Octokit.NotFoundException e)
+            {
+                Logger.ErrorLocalized(nameof(Resources.Error_Prefix), e.Message);
+                Logger.ErrorLocalized(nameof(Resources.OctokitNotFound_Error));
+                return false;
+            }
+            finally
+            {
+                TelemetryManager.Log.WriteEvent(commandEvent);
+            }
+        }
+
+        private static void ShowAllManifests(Manifests manifests)
+        {
+            DisplayInstallerManifest(manifests.InstallerManifest);
+            DisplayDefaultLocaleManifest(manifests.DefaultLocaleManifest);
+            DisplayLocaleManifests(manifests.LocaleManifests);
+            DisplayVersionManifest(manifests.VersionManifest);
+        }
+
+        private static void DisplayInstallerManifest(InstallerManifest installerManifest)
+        {
+            Logger.InfoLocalized(nameof(Resources.InstallerManifest_Message));
+            Console.WriteLine(installerManifest.ToYaml(true));
+        }
+
+        private static void DisplayVersionManifest(VersionManifest versionManifest)
+        {
+            Logger.InfoLocalized(nameof(Resources.VersionManifest_Message));
+            Console.WriteLine(versionManifest.ToYaml(true));
+        }
+
+        private static void DisplayDefaultLocaleManifest(DefaultLocaleManifest defaultLocaleManifest)
+        {
+            Logger.InfoLocalized(nameof(Resources.DefaultLocaleManifest_Message), defaultLocaleManifest.PackageLocale);
+            Console.WriteLine(defaultLocaleManifest.ToYaml(true));
+        }
+
+        private static void DisplayLocaleManifests(List<LocaleManifest> localeManifests)
+        {
+            foreach (var localeManifest in localeManifests)
+            {
+                Logger.InfoLocalized(nameof(Resources.LocaleManifest_Message), localeManifest.PackageLocale);
+                Console.WriteLine(localeManifest.ToYaml(true));
+            }
+        }
+
+        private static void DisplaySingletonManifest(SingletonManifest singletonManifest)
+        {
+            Logger.InfoLocalized(nameof(Resources.SingletonManifest_Message));
+            Console.WriteLine(singletonManifest.ToYaml(true));
+        }
+
+        private void ParseArgumentsAndShowManifest(Manifests manifests)
+        {
+            if (manifests.SingletonManifest != null)
+            {
+                DisplaySingletonManifest(manifests.SingletonManifest);
+                return;
+            }
```

Comment:
I think this is a good as the default behavior if no args are provided, but having this here will prevent all the other arguments from working. You may want to have a boolean that checks whether all flags are false before displaying this.

Reply:
> but having this here will prevent all the other arguments from working.

I'm not sure I understand this 👀 My thinking behind adding it before we check for the arguments is that as a user a command like

```
wingetcreate show NSIS.NSIS --installer-manifest --defaultlocale-manifest
```

should not fail even though the manifest is a singleton manifest. If the user is doing `--installer-manifest --defaultlocale-manifest`, the user probably wants to see the installer nodes and fields like PackageName, Publisher etc. Instead of saying to user that hey this is a singleton manifest and the args are not supported or only printing the singleton if no args are provided, having the check above makes it so that the args have no effect and the singleton manifest is printed anyways and the metadata the user is looking for is displayed.
If the check is moved downwards, we probably have to do null checks for installer, defaultlocale, locale manifests etc. in the case that the package has a singleton manifest

Reply:
So I was doing some testing with my own build and had it pointed to `Microsoft/winget-pkgs-submission-test` repo and it was pulling down a singleton, which I didn't realize so I thought it was always displaying singleton.

Your explanation makes sense so no need to make any changes :)